src/List.mo

```diff
@@ -1569,18 +1756,35 @@ module {
   ///
   /// *Runtime and space assumes that `compare` runs in O(1) time and space.
   public func min<T>(list : List<T>, compare : (T, T) -> Order.Order) : ?T {
-    if (isEmpty(list)) return null;
-
-    var minSoFar = at(list, 0);
-    forEach<T>(
-      list,
-      func(x) = switch (compare(x, minSoFar)) {
-        case (#less) minSoFar := x;
-        case _ {}
-      }
-    );
+    var minSoFar : T = switch (first(list)) {
+      case (?x) x;
+      case null return null
+    };
+
+    let blocks = list.blocks;
+    let blockCount = blocks.size();
+
+    var i = 1;
+    while (i < blockCount) {
+      let db = blocks[i];
+      let sz = db.size();
+      if (sz == 0) return ?minSoFar;
+
+      var j = 0;
+      while (j < sz) {
+        switch (db[j]) {
+          case (?x) switch (compare(x, minSoFar)) {
+            case (#less) minSoFar := x;
+            case _ {}
+          };
+          case null return ?minSoFar
+        };
+        j += 1
+      };
+      i += 1
+    };
```

Comment:
Are we comparing here the first element against itself? Is this intentional to simplify the code?